src/ArtisanApartmentCommands.php

```diff
@@ -0,0 +1,176 @@
+        }
+
+        if (!$this->schema->doesSchemaExist($schemaName)) {
+            throw new SchemaDoesntExistException('The apartment ' . $schemaName . ' does not exist.');
+        }
+
+        app('db')->transaction(function () use ($schemaName) {
+            $this->deleteSchema($schemaName);
+        });
+    }
+
+    /**
+     * Create the schema.
+     *
+     * @param $schemaName
+     * @throws SchemaNameNotValidException
+     */
+    protected function createSchema($schemaName)
+    {
+        if (!ApartmentHelpers::isSchemaNameValid($schemaName)) {
+            throw new SchemaNameNotValidException('The apartment ' . $schemaName . ' is not valid. It must be all lowercase and only contain letters, numbers, or underscores.');
+        }
+        app('db')->statement('CREATE SCHEMA ' . $schemaName);
+    }
+
+    /**
+     * Drop the schema.
+     *
+     * @param $schemaName
+     * @throws SchemaNameNotValidException
+     */
+    protected function deleteSchema($schemaName)
+    {
+        if (!ApartmentHelpers::isSchemaNameValid($schemaName)) {
+            throw new SchemaNameNotValidException('The apartment ' . $schemaName . ' is not valid. It must be all lowercase and only contain letters, numbers, or underscores.');
+        }
+        app('db')->statement('DROP SCHEMA ' . $schemaName . ' CASCADE');
+    }
+
+    /**
+     * Create the schema's migration table which will mimic the public migration table.
+     *
+     * @param $schemaName
+     */
+    protected function createSchemaMigrationTable($schemaName)
+    {
+        LumenSchema::create(ApartmentHelpers::getSchemaTableFormat($schemaName, 'migrations'), function ($table) {
+            $table->increments('id');
+            $table->string('migration');
+        });
+    }
+
+    /**
+     * Create the schema's metadata table.
+     *
+     * @param $schemaName
+     */
+    protected function createSchemaMetadata($schemaName)
+    {
+        LumenSchema::create(ApartmentHelpers::getSchemaTableFormat($schemaName, 'apartment_metadata'),
+            function ($table) {
+                $table->string('name');
+                $table->integer('created_by');
+                $table->integer('created_at');
+            });
+    }
+
+    /**
+     * Update schema's metadata.
+     *
+     * @param $schemaName
+     * @param $createdBy
+     * @param $createdAt
+     */
+    protected function updateSchemaMetadata($schemaName, $createdBy, $createdAt)
+    {
+        app('db')->table(ApartmentHelpers::getSchemaTableFormat($schemaName, 'apartment_metadata'))->insert([
+            'name' => $schemaName,
+            'created_by' => $createdBy,
+            'created_at' => $createdAt,
+        ]);
+    }
+
+    /**
+     * Check if the schemaName is public.
+     *
+     * @param $schemaName
+     * @return bool
+     */
+    protected function isSchemaPublic($schemaName)
+    {
+        if ($schemaName != self::PUBLIC_SCHEMA) {
```

Comment:
Can't this just be `return $schemaName == self::PUBLIC_SCHEMA;`?  If-statements/ternaries with boolean returns are weird.